File: leetcode/0038/countAndSay.swift
func countAndSay(_ n: Int) -> String {
    var arr = ["1"]
    if n == 1 {
        return arr.joined()
    }
    for _ in 1..<n {
        var word = arr[0]
        var num = 1
        var temp: String = ""
        for j in 1..<arr.count {
            if word == arr[j] {
                num += 1
            } else {
                let numChar = "\(num)"
                temp += numChar
                temp += word
                
                word = arr[j]
                num = 1
            }
        }
        let numChar = "\(num)"
        temp += numChar
        temp += word
        arr = Array(temp).map({ c in
            String(c)
        })
    }
    return arr.joined()
}

print(countAndSay(4))

File: leetcode/0038/countAndSay.py
def countAndSay(n: int) -> str:
	arr = "1"
	if n == 1:
		return arr
	for i in range(n):
		if i == 0:
			continue
		word = arr[0]
		num = 1
		temp = ""
		for j in range(len(arr)):
			if j == 0:
				continue
			if word == arr[j]:
				num += 1
			else:
				num_char = str(num)
				temp += num_char
				temp += word

				word = arr[j]
				num = 1
		num_char = str(num)
		temp += num_char
		temp += word
		arr = temp
	return arr

print(countAndSay(5))